src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -149,16 +157,66 @@ private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
                 return variableDeclaration.Type;
             }
 
+            // x = null;
+            if (node.Parent is AssignmentExpressionSyntax assignment)
+            {
+                var symbol = model.GetSymbolInfo(assignment.Left).Symbol;
+                if (symbol is ILocalSymbol local)
+                {
+                    var syntax = local.DeclaringSyntaxReferences[0].GetSyntax();
+                    if (syntax is VariableDeclaratorSyntax declarator &&
+                        declarator.Parent is VariableDeclarationSyntax declaration &&
+                        declaration.Variables.Count == 1)
+                    {
+                        return declaration.Type;
+                    }
+
+                }
+                else if (symbol is IParameterSymbol parameter)
```

Comment:
We don't want to support fields/properties?

Reply:
Not sure why I didn't ;-) Added now. Thanks